packages/botonic-plugin-watson/src/index.js

```diff
@@ -1,36 +1,99 @@
-import IamAuthenticator from 'ibm-watson/auth'
+import axios from 'axios'
+
+import WatsonOutputParser from './watson-output-parser'
 
 export default class BotonicPluginWatson {
+  REQUIRED_OPTION_FIELDS = ['apikey', 'url', 'assistant_id']
+
+  EXPECTED_SESSION_ID_REQUEST_STATUS = 201
+  EXPECTED_OUTPUT_REQUEST_STATUS = 200
+
   constructor(options) {
-    this.options = options
+    try {
+      this.validateOptions(options)
+      this.init(options)
+    } catch (e) {
+      console.error(String(e))
+    }
+  }
+
+  validateOptions(options) {
+    if (!options) throw new Error('No options provided.')
+    if (!this.areRequiredOptionsProvided(options))
+      throw new Error('Missing one or more fields.')
+    if (!this.areRequiredOptionsDefined(options))
+      throw new Error('One or more fields are not defined.')
+  }
+
+  areRequiredOptionsProvided(options) {
+    const requiredOptionsFieldsProvided = Object.keys(options).filter(field =>
+      this.REQUIRED_OPTION_FIELDS.includes(field)
+    )
+    return (
+      requiredOptionsFieldsProvided.length == this.REQUIRED_OPTION_FIELDS.length
+    )
+  }
+
+  areRequiredOptionsDefined(options) {
+    const requiredOptionsProvided = Object.entries(
+      options
+    ).filter(([field, _]) => this.REQUIRED_OPTION_FIELDS.includes(field))
+    return !requiredOptionsProvided.some(([_, value]) => value === undefined)
+  }
+
+  init(options) {
+    this.apiKey = options.apikey
+    this.url = options.url
+    this.assistantId = options.assistant_id
+    this.version = options.version || '2020-04-01'
+    this.sessionId = this.getSessionId()
+  }
+
+  async getSessionId() {
+    const { data } = await axios.post(
+      `${this.getAssistantEndpoint()}/sessions?version=${this.version}`,
+      {},
+      {
+        auth: this.getAuthentication(),
+        validateStatus: status =>
+          status === this.EXPECTED_SESSION_ID_REQUEST_STATUS,
+      }
+    )
+    return data.session_id
   }
 
-  async pre({ input, session, lastRoutePath }) {
-    let intent = null
-    let confidence = 0
-    let intents = []
-    let entities = []
+  getAssistantEndpoint() {
+    return `${this.url}/v2/assistants/${this.assistantId}`
+  }
+
+  getAuthentication() {
+    return { username: 'apikey', password: this.apiKey }
+  }
 
+  async pre({ input }) {
     try {
-      const assistant = new AssistantV1({
-        authenticator: new IamAuthenticator({ apikey: this.options.apiKey }),
-        url: this.options.url,
-        version: '2018-09-19',
-      })
-
-      const res = await assistant.message({
-        input: { text: input.data },
-        workspaceId: this.options.workspaceId,
-      })
-      intent = res.result.intents[0].intent
-      confidence = res.result.intents[0].intent.confidence
-      intents = res.result.intents
-      entities = res.result.entities
-    } catch (error) {
-      console.log(error)
+      const output = await this.getWatsonOutput(input.data)
```

Comment:
if initialization fails, here you will get strange errors.
I suggest that here you check eg sessionId and if it's undefined, you log st like "Not executing Watson because initialization failed"